### PR TITLE
Add support for API to fail if returned user is not same as passed in

### PR DESCRIPTION
**Pull request by kpanwar**
Add support for API to fail if returned user is not same as passed in. per #281

### CODE REVIEW

File: src/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 1]
```diff
@@ -104,11 +104,11 @@ internal TokenCache UserTokenCache
             }
         }
 
-        /// <summary>
+/*        /// <summary>
```
Comment (weijjia) [inline src/Microsoft.Identity.Client/ClientApplicationBase.cs]: So you're hiding it by commenting it out? Can we just remove it? If we get feedback from developers to add it back, we can always find it in git commits and copy it back :)

File: src/Microsoft.Identity.Client/IAuthenticationResult.cs [hunk 2]
```diff
@@ -68,5 +68,10 @@ public interface IAuthenticationResult
         /// Gets the scope values returned from the service.
         /// </summary>
         IEnumerable<string> Scope { get; }
+
```
Comment (weijjia) [inline src/Microsoft.Identity.Client/IAuthenticationResult.cs]: Ok, let's follow up on Monday. I think both ios and android only return accessToken, expiresOn, User, and scope back.

File: src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs [hunk 3]
```diff
@@ -140,6 +139,16 @@ private Uri CreateAuthorizationUri(bool addVerifier = false, bool addState = fal
                 requestParameters[OAuth2Parameter.State] = _state;
             }
 
+            //add uid/utid values to QP if user object was passed in.
+            if(AuthenticationRequestParameters.User != null)
+            {
+                requestParameters[OAuth2Parameter.LoginHint] = AuthenticationRequestParameters.User.DisplayableId;
+                AuthenticationRequestParameters.ClientInfo = ClientInfo.CreateFromEncodedString(AuthenticationRequestParameters.User.Identifier);
+                requestParameters[OAuth2Parameter.LoginReq] = AuthenticationRequestParameters.ClientInfo.UniqueIdentifier;
```
Comment (weijjia) [inline src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs]: what if uid and utid on user object is empty?

**Comment by weijjia** [comment]
You didn't answer the question why we're commenting out the correlation id setter instead of removing the code. :) It's minor tho, feel free to merge.

**Comment by kpanwar** [comment]
fixed

---
In reply to: [292753870](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/305#issuecomment-292753870) [](ancestors = 292753870)

**Pull request merged**